### PR TITLE
[slos] fix API payloads

### PR DESCRIPTION
**Pull request by platinummonkey**
We changed the name of this the threshold `slo` property to `target` as it's more meaningful.  The hardest part of SW - naming.

Also updated the content types of create and added request validation as well.

### CODE REVIEW

**Comment by bkabrda** [comment]
The recent changes have removed the `createSLOThreshold` and `createUpdateServiceLevelObjective`, but (some of) the accessors for these remained in the autogenerated accessors file for these, so the compilation fails. Could you please remove these so I can test and merge? Thanks!

**Comment by platinummonkey** [comment]
yup! I'm waiting for us to merge && deploy the final API changes before continuing this work 👌

**Comment by bkabrda** [comment]
@platinummonkey ok, feel free to ping me whenever this is ready to be re-reviewed.

**Comment by platinummonkey** [comment]
@bkabrda good to go 👍 

```
--- PASS: TestServiceLevelObjectivesCreateGetUpdateAndDelete (3.84s)
PASS
```

File: integration/service_level_objectives_test.go [hunk 1]
```diff
@@ -34,32 +35,38 @@ func TestServiceLevelObjectivesCreateGetUpdateAndDelete(t *testing.T) {
 	assert.Equal(t, expected.Description, actual.Description)
 	assert.True(t, expected.Thresholds.Equal(actual.Thresholds))
 
+	time.Sleep(time.Second)
```
Comment (bkabrda) [inline integration/service_level_objectives_test.go]: Why the sleep? Is it expected that the SLO object is not available through API immediately? Or is it available but incomplete?

Comment (platinummonkey) [reply integration/service_level_objectives_test.go]: I found that in a few rare cases, the read replicas were behind just enough to make this fail without some artificial delay.

Comment (bkabrda) [reply integration/service_level_objectives_test.go]: Ok, thanks for the explanation.

**Comment by bkabrda** [comment]
LGTM now, merging. Thanks!

**Pull request merged**